server/client.go

```diff
@@ -1833,6 +1835,68 @@ func (c *client) processPong() {
 	}
 }
 
+// Header pubs take form HPUB <subject> [reply] <hdr_len> <total_len>\r\n
+func (c *client) processHeaderPub(arg []byte) error {
+	// Unroll splitArgs to avoid runtime/heap issues
```

Comment:
right around here I would expect a check 
if c.srv.opts.NoHeaderSupport {
    c.sendErr(ErrMsgHeadersNotSupported)
    c.closeConnection(MsgHeaderViolation)
    return ErrMsgHeadersNotSupported
}

Reply:
Yes agree, but again removed all of the old tests from a real client, this is all just very low level, but will get it there.

Reply:
Actually that is the c.headers from above so I don't have to grab a server lock etc..